src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,428 @@ class Guides::Database::Querying < GuideAction
+
+    # returns nil if not found
+    user = UserQuery.new.last?
+    ```
+
+    ### Select by primary key
+
+    Selecting the user with `id = 3`.
+    `SELECT COLUMNS FROM users WHERE users.id = 3 LIMIT 1`
 
     ```crystal
-    # Lucky parses the string and use an Int32 in the query
-    UserQuery.new.age("30")
+    # raise Avram::RecordNotFound if nil
+    user = UserQuery.new.find(3)
+    ```
+
+    ### Select distinct / distinct on
 
-    # Won't accept other types
-    # Fails at compile time because you can't pass a Float64
-    UserQuery.new.age(30.0)
+    `SELECT DISTINCT COLUMNS FROM users`
+
+    ```crystal
+    UserQuery.new.distinct
     ```
 
-    > Note that this will change in the future but this is the current behavior. In
-    the future only the defined type (`Int32` in this example) will work. You will
-    then need to manually parse it. Don't worry though, Lucky will add some helpers
-    to actions and the model to make this easy and maintainable.
+    Select distinct rows based on the `name` column `SELECT DISTINCT ON (users.name) FROM users`
 
-    ## Ordering results
+    ```crystal
+    UserQuery.new.distinct_on(&.name)
+    ```
 
-    ````crystal
-    UserQuery.new.age.desc_order
-    # or
-    UserQuery.new.age.asc_order
-    ````
+    ## Where Queries
 
-    ## Chaining methods
+    The `WHERE` clauses are the most common used in SQL. Each of the columns generated by the model
+    will give you a method for running a `WHERE` on that column. (e.g. the `age` column gives us
+    `WHERE age = value` with the `age(value)` method.
 
-    You can chain where clauses and most other query methods.
+    ### A = B
+
+    Find rows where `A` is equal to `B`.
+
+    `SELECT COLUMNS FROM users WHERE users.age = 54`
 
     ```crystal
-    UserQuery.new.age.gt(28).age.desc_order
+    UserQuery.new.age(54)
     ```
 
-    ## Running the query
+    ### A = B AND C = D
 
-    The query does not actually hit the database until a method is called to fetch a
-    result or iterate over results.
+    Find rows where `A` is equal to `B` and `C` is equal to `D`.
 
-    Some of the most common methods are:
+    `SELECT COLUMNS FROM users WHERE users.age = 43 AND users.admin = true`
 
-    * `first`
-    * `find`
-    * `each`
+    ```crystal
+    UserQuery.new.age(43).admin(true)
+    ```
 
-    For example:
+    > All query methods are chainable!
+
+    ### A != B
+
+    Find rows where `A` is not equal to `B`.
+
+    `SELECT COLUMNS FROM users WHERE users.name != 'Billy'`
 
     ```crystal
-    # The query is not yet run
-    query = UserQuery.new
-    query.name("Sally")
-    query.age(30)
+    UserQuery.new.name.not.eq("Billy")
+    ```
 
-    # The query will run once `each` is called
-    # Results are not cached so a request will be made every time you call `each`
-    query.each do |user|
-      pp user.name
-    end
+    > The `not` method can be used to negate other methods like `eq`, `gt`, `lt`, `in`.
+
+    ### A gt/lt B
```

Comment:
Yeah, it is sort of a lot to take in. The benefit this way though is being able to search for them individually. Like searching "less than". I'm not sure how that search works though. Maybe it would pick up the code examples if I used crystal comments?

```crystal
# less than or equal to `WHERE users.age <= 12`
UserQuery.new.age.lte(12)

# less than `WHERE users.updated_at < '#{3.months.ago}'`
UserQuery.new.updated_at.lt(3.months.ago)
```

Reply:
That could work. You could list them before the examples:

> Lucky supports inequality queries like less than, greater than, and their or-equals counter parts

```crystal
# WHERE users.age <= 12
UserQuery.new.age.lte(12)

# WHERE users.updated_at < '#{3.months.ago}'
UserQuery.new.updated_at.lt(3.months.ago)
```